OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -502,9 +502,9 @@ bool HasAdequateAirUnitReloadBuildings(ActorInfo actorInfo)
 						.Where(a => resourceTypeIndices.Get(Map.GetTerrainIndex(a)))
 						.Shuffle(Random);
 
-					foreach (var c in nearbyResources)
+					if (nearbyResources.Any())
 					{
-						var found = findPos(c, baseCenter, 0, Info.MaxBaseRadius);
+						var found = findPos(nearbyResources.First(), baseCenter, 0, Info.MaxBaseRadius);
```

Comment:
This will enumerate the nearby resources twice. If you're just going to pick any random nearby resource try:
```c#
var randomNearbyResource = Map.FindTilesInCircle(baseCenter, Info.MaxBaseRadius)
    .Where(a => resourceTypeIndices.Get(Map.GetTerrainIndex(a)))
    .RandomOrDefault(Random);
```